### DISCUSSION TITLE
The way the candidates appear in Golang lsp-mode is strange.

### DISCUSSION THREAD

**Discussion by kamuycikap9821** [emacs-lsp/lsp-mode#4061] [Q&A]
MacOS Ventura
Emacs28.2

Please see here.
Below is the behavior of lsp-mode in the Go language source code.

■Linux
https://gyazo.com/7d43aedfae0b8881191e9ae56d9b6d50

■Mac
https://gyazo.com/98cb9ecc64199aa9388e357a0082da39

When I hit the p key, candidates for Printf, etc. appear on Linux, but not on Mac.
Am I the only one with this behavior?
Are you guys experiencing similar symptoms with Ventura's Emacs 28.2?

■my setup
(when (require 'lsp-mode nil t)

  (define-key lsp-mode-map (kbd "M-.") #'lsp-find-definition)
  (define-key lsp-mode-map (kbd "M-?") #'lsp-find-references)
  (define-key lsp-mode-map (kbd "M-,") #'xref-pop-marker-stack)
  (define-key lsp-mode-map (kbd "M-/") #'lsp-execute-code-action)
  (define-key lsp-mode-map (kbd "C-c C-d") #'lsp-describe-thing-at-point) 
  (define-key lsp-mode-map (kbd "C-c C-r") #'lsp-rename)
 ;(define-key lsp-mode-map (kbd "C-c C-f") #'lsp-format-buffer)
  (define-key lsp-mode-map (kbd "M-g g") #'consult-goto-line)

  (setq lsp-enable-snippet nil) 
  (setq lsp-completion-provider :capf)
  )

;; lsp-ui
(when (require 'lsp-ui nil t)
  (setq lsp-ui-doc-enable t)
  (setq lsp-ui-doc-header t)
  (setq lsp-ui-doc-include-signature t)
  (setq lsp-ui-doc-max-width 150)
  (setq lsp-ui-doc-max-height 30)
  (setq lsp-ui-peek-enable t)

  ;; lsp-ui hook
  (add-hook 'lsp-mode-hook 'lsp-ui-mode)
  )

;; dap-mode
(when (require 'dap-mode nil t)
  (dap-mode 1)
  (dap-auto-configure-mode 1)
  (dap-ui-mode 1)
  (dap-tooltip-mode 1)
  (tooltip-mode 1)
  (require 'dap-hydra)
  (require 'dap-gdb-lldb)
  (dap-gdb-lldb-setup)
  )

;; go-mode
(when (require 'go-mode nil t)

  (add-to-list 'auto-mode-alist '("\\.go\\'" . go-mode))

  (setq auto-mode-alist
        
        (append
         '(("\\.go$" . go-mode))
         auto-mode-alist))

  ;; HOOK
  (add-hook 'go-mode-hook
            (lambda ()
              (setq-default)

              (setq tab-width 4)
              (setq standard-indent 4)
              (setq indent-tabs-mode nil)

              ;; Golang Debugger
              (require 'go-dlv)

              (lsp)
              (lsp-deferred)
              ;; imenu
              (go-imenu-setup)

              (message "Go-mode add-hook ---end")
              )
            )
  )

**Comment by yyoncho** [comment]
Apart from different server versions, this could happen depending on whether the filtering is performed client or server side. Note that if the server returns the full result we won't call it and we will perform a fuzzy match client side. On the other hand, if the server does not return in time then we will call the server again and it may use a different strategy for filtering.

**Reply by kamuycikap9821** [reply]
Is there anything I can do?
